backend/database.js

```diff
@@ -51,6 +51,52 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'backend/db/streamEntriesBefore': async function (contractID: string, before: string, limit: number): Promise<*> {
+    let currentHEAD = await sbp('chelonia/db/latestHash', contractID)
+    if (!currentHEAD) {
+      throw Boom.notFound(`contractID ${contractID} doesn't exist!`)
+    }
```

Comment:
Why is this retrieving the latestHash for the contract? It's unclear to me why that's necessary. You just need to start at `before` and go backwards, right?

Reply:
Correct. I also think it is weird. I need to check it again. I am not sure why I did it this way. :grin: